src/realm/table.cpp

```diff
@@ -2217,21 +2216,13 @@ Obj Table::create_object(ObjKey key, const FieldValues& values)
     if (m_primary_key_col)
         throw IllegalOperation(util::format("Table has primary key: %1", get_name()));
     if (key == null_key) {
-        GlobalKey object_id = allocate_object_id_squeezed();
-        key = object_id.get_local_key(get_sync_file_id());
-        // Check if this key collides with an already existing object
-        // This could happen if objects were at some point created with primary keys,
-        // but later primary key property was removed from the schema.
-        while (m_clusters.is_valid(key)) {
-            object_id = allocate_object_id_squeezed();
-            key = object_id.get_local_key(get_sync_file_id());
-        }
-        if (auto repl = get_repl())
```

Comment:
was this a bug before? (i.e, the instruction was replicated only if the key was null)

Reply:
You could say it was a bug, but calling with a specific key was only used in testing.